### ISSUE TITLE
Crash in Record_Get

### ISSUE THREAD

**Issue by TransAMrit** [FalkorDB/FalkorDB#807]
Hi, the following query crashes my falkordb with 100% consistency:

```
CALL {
MATCH (cdo:C) WHERE cdo.persona_id = 3 RETURN cdo ORDER BY cdo.create_time DESC
UNION ALL
MATCH (cdo:E) WHERE cdo.persona_id = 3 RETURN cdo ORDER BY cdo.create_time DESC
}
RETURN cdo LIMIT 100
```

If I add LIMIT 100 at the end of both ORDER BY's, the crash goes away. However, I only have 723 vertices in this test server.

Crash log is attached.
[falkor.crash.log](https://github.com/user-attachments/files/17369363/falkor.crash.log)

**Comment by gkorland** [comment]
@TransAMrit thanks for reporting the issue, any chance you share the dataset that recreates this issue?

**Comment by TransAMrit** [comment]
> @TransAMrit thanks for reporting the issue, any chance you share the dataset that recreates this issue?

I can't share the exact dataset unfortunately, but I can look into Falkor's export/import functionality and see if I can munge the dataset a bit while still reproducing. Hopefully I can get to that sometime this week.

**Comment by swilly22** [comment]
@TransAMrit I think we can do without a dataset.
In the mean time let me suggest a similar alternative to the query you're running:
```
MATCH (cdo:C) WHERE cdo.persona_id = 3 
WITH cdo
ORDER BY cdo.create_time DESC
LIMIT 100
WITH collect(cdo) as Cs
MATCH (cdo:E) WHERE cdo.persona_id = 3
WITH cdo, Cs
ORDER BY cdo.create_time DESC
LIMIT 100
WITH collect(cdo) as Es, Cs
WITH Cs + Es AS cdos
RETURN cdos[..100]
```

**Comment by TransAMrit** [comment]
> @TransAMrit I think we can do without a dataset. In the mean time let me suggest a similar alternative to the query you're running:

Oh thanks, interesting search! Is this a preferred form for this query, or instead just an alterative to avoid the crash? The crash is completely avoided if include LIMIT for each of the two inner MATCHes, and the original search looks simpler and of similar performance (2x Node By Label Scan)...

**Comment by swilly22** [comment]
I agree, if you can avoid the crash by introducing LIMITs than your original query is simpler to reason with. in case you're getting similar performance on both stick with the simpler query.

**Comment by swilly22** [comment]
Steps to reproduce:
```
GRAPH.QUERY g "UNWIND range (0, 10000) AS x CREATE (:C {persona_id:3, create_time:x}), (:E {persona_id:3, create_time:x})"

GRAPH.QUERY g "CALL { MATCH (cdo:C) WHERE cdo.persona_id = 3 RETURN cdo ORDER BY cdo.create_time DESC UNION ALL MATCH (cdo:E) WHERE cdo.persona_id = 3 RETURN cdo ORDER BY cdo.create_time DESC } RETURN cdo LIMIT 100"
```

**Comment by DvirDukhan** [comment]
Closing as no longer reproducible.

This was re-tested on the current stable release, **v4.20.1**
(`falkordb/falkordb:latest`), and the reported behaviour no longer occurs:

**Reproduction**

```cypher
UNWIND range (0, 10000) AS x CREATE (:C {persona_id:3, create_time:x}), (:E {persona_id:3, create_time:x})

CALL { MATCH (cdo:C) WHERE cdo.persona_id = 3 RETURN cdo ORDER BY cdo.create_time DESC UNION ALL MATCH (cdo:E) WHERE cdo.persona_id = 3 RETURN cdo ORDER BY cdo.create_time DESC } RETURN cdo LIMIT 100
```

| Build | Result |
| --- | --- |
| `falkordb/falkordb:latest` (v4.20.1) | No crash. UNION ALL of two ORDER BY subqueries inside CALL{} + outer LIMIT 100 returns 100 correctly-ordered rows (descending create_time), container uptime unaffected. |

Verified via docker inspect StartedAt and docker logs that neither container restarted/crashed across the run. Used the maintainer's own confirmed minimal reproduction (20002 nodes split across labels :C/:E) rather than the original reporter's private dataset. Since the crash does not reproduce on falkordb:latest (v4.20.1), it was already fixed in the C engine and cannot be credited to the Rust migration, even though edge-rs also handles it correctly.

The defect appears to have been fixed in the C engine at some point between the
version this was reported against and v4.20.1. If you can still reproduce it on
v4.20.1 or later, please reopen with the exact version and a repro.
